packages/ui-markdown-editor/src/FormattingToolbar/index.js

```diff
@@ -54,11 +54,36 @@ const FormattingToolbar = ({
       const rect = domRange.getBoundingClientRect();
       const CARET_TOP_OFFSET = 15;
       el.style.opacity = 1;
-      el.style.top = `${rect.top + rect.height + window.pageYOffset + CARET_TOP_OFFSET}px`;
-      el.style.left = `${rect.left
-          + window.pageXOffset
-          - el.offsetWidth / 2
-          + rect.width / 2}px`;
+      el.style.top = `${
+        rect.top + rect.height + window.pageYOffset + CARET_TOP_OFFSET
+      }px`;
+      let w = el.children[0],
+        z = el.parentElement;
+      let calPos = rect.left  - el.offsetWidth / 2  ;
+
+
+      // When the modal goes off page from left side
+      if (calPos < 0) {
+        // start from 10px 
+        calPos = 10;
+        w.style.left = `${rect.left - 10}px`;
+      }
+
+      // calculate the endpoint of the modal
+      let rightEndPos = calPos + el.offsetWidth,
+        containerWidth = z.offsetWidth;
+
+      // When the modal goes off the page from right side
+      if (rightEndPos > containerWidth) {
+        let diff = rightEndPos-containerWidth;
+        // extra space of 10px on right side to look clean
+        diff+=10;
+        calPos=calPos-diff;
+        let shift=diff-5;
+        w.style.left= `calc(50% + ${shift}px)`;
+      }
+
+      el.style.left = `${calPos}px`;
```

Comment:
This looks great!!

Some semantic nitpicks:
- Could you name `w` in `let w = el.children[0];` more descriptively or describe in a comment?
- `z = el.parentElement;` doesn't seem necessary, as later you could just do:
  `let containerWidth = el.parentElement.offsetWidth;`
- Could you name `calPos` in `let calPos = rect.left  - el.offsetWidth / 2;` more descriptively or describe in a comment?
- Some of these variables don't change and thus can be `const`:
`const rightEndPos = calPos + el.offsetWidth;`
`const containerWidth = z.offsetWidth;`

Reply:
@irmerk 
I have made a PR with the above suggestions implemented.
Please let me know if there are any changes required :)